src/main/java/ai/rev/speechtotext/models/asynchronous/RevAiJobOptions.java

```diff
@@ -141,11 +187,40 @@ public String getCallbackUrl() {
    * property is optional.
    *
    * @param callbackUrl The url to POST to when job processing is complete.
+   * @deprecated Use setNotificationConfig instead
    */
   public void setCallbackUrl(String callbackUrl) {
     this.callbackUrl = callbackUrl;
   }
 
+  /**
+   * Returns the notification config object.
+   *
+   * @return the notification config.
+   */
+  public CustomerUrlData getNotificationConfig() {
+    return notificationConfig;
+  }
+
+  /**
+   * Optional property to specify the callback url that Rev AI will POST to when job processing is complete
+   *
+   * @param callbackUrl The url to POST to when job processing is complete.
+   * @param authHeaders Optional parameter to authenticate access to the callback url
+   */
+  public void setNotificationConfig(String callbackUrl, Map<String, String> authHeaders) {
+    this.notificationConfig = new CustomerUrlData(callbackUrl, authHeaders);
+  }
+
+  /**
+   * Optional property to specify the callback url that Rev AI will POST to when job processing is complete
+   *
+   * @param callbackUrl The url to POST to when job processing is complete.
+   */
+  public void setNotificationConfig(String callbackUrl) {
+    this.notificationConfig = new CustomerUrlData(callbackUrl, null);
```

Comment:
I think standard in java is to call the overloaded method with some parameters null. This way there's one "source of truth" method and only one place needs to be changed if updates happen.

In this case I don't think it REALLY matters since its just a setter so I'll leave it up to you

Reply:
I can't find a great source on that, but I think it's less elegant to expect the user to have to pass in nulls even if it is a bit harder to support. Most things I'm seeing for optional params in java suggest the overloaded method approach. And since we've only got one optional param something like a builder pattern is overkill imo.

Reply:
I think you misunderstood my comment. I wasn't saying the user should be the one passing nulls. I was saying this method specifically should call setNotificationConfig(callbackUrl, null) instead of duplicating the logic

Reply:
Oh I see what you mean, I completely missed that. Totally agree.